### PR TITLE
[YUNIKORN-2798] MaxApplications should support static queues

### PR DESCRIPTION
**Pull request by zhuqi-lucas**
…namic queue.

### What is this PR for?
When developing the set max app annotation, i found there are something broken logic for setting max apps for static queue, we should improve and fix it.

Details:
https://github.com/apache/yunikorn-core/pull/927#discussion_r1705132945

### What type of PR is it?
* [x] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-2798
* Put link here, and add [YUNIKORN-*Jira number*] in PR title, eg. `[YUNIKORN-2] Gang scheduling interface parameters`

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/queue_test.go [hunk 1]
```diff
@@ -2015,6 +2016,7 @@ func TestApplyConf(t *testing.T) {
 	assert.Assert(t, leaf.template == nil)
 	assert.Assert(t, leaf.maxResource != nil)
 	assert.Assert(t, leaf.guaranteedResource != nil)
+	assert.Assert(t, leaf.maxRunningApps == 100)
```
Comment (ryankert01) [inline pkg/scheduler/objects/queue_test.go:2019]: Would `assert.Equal()` be better in this case? It can provide more detail when failed.

Comment (zhuqi-lucas) [reply pkg/scheduler/objects/queue_test.go]: Thank you @ryankert01 for review, addressed in latest PR.

**Review by ryankert01 — APPROVED** [review]
+1, LGTM

**Review by pbacsko — APPROVED** [review]
LGTM

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**